shell/platform/android/io/flutter/plugin/editing/ListenableEditingState.java

```diff
@@ -240,6 +276,24 @@ public final int getComposingEnd() {
     return BaseInputConnection.getComposingSpanEnd(this);
   }
 
+  @Override
+  public void setSpan(Object what, int start, int end, int flags) {
+    super.setSpan(what, start, end, flags);
+    // Setting a span does not involve mutating the text value in the editing state. Here we create
+    // a non text update delta with any updated selection and composing regions.
+    mBatchTextEditingDeltas.add(
```

Comment:
It is possible but it shouldn't affect the final result since they are `nonTextUpdates`. In the end the final `TextEditingDelta` in the batch will have the most up to date selection/composing regions. In theory if `setSpan` is called in `replace` we will want to capture that anyways.

Reply:
The batch may look something like this if `super.replace()` calls `setSpan`.
`[nonTextUpdate, nonTextUpdate, nonTextUpdate, delta]`
In this case the last delta will still have the most up to date composing/selection regions.

Reply:
Ideally there should be no overlap between consecutive deltas and there should be no redundant deltas. Should be fine for now.